### DISCUSSION TITLE
unsigned value types as component fields

### DISCUSSION THREAD

**Discussion by sergiyha** [friflo/Friflo.Json.Fliox#37] [General]
Hi. I am testing the ECS engine with Unity and stuck with some feature/bugs. Each time I create the new entt store having at least one component with unsigned value type as struct field,  an invalid operation exception gets thrown. Is it ok and will components support it in the future?
![image](https://github.com/friflo/Friflo.Json.Fliox/assets/16262019/fe86fa87-2ebf-4fba-be51-710c5f830444)

**Comment by friflo** [comment]
Hi,
you are right.

Currently only CLS compliant simply types are supported. See: [Simple types](https://learn.microsoft.com/de-de/dotnet/csharp/language-reference/language-specification/types#835-simple-types)
Non compliant CLS types are currently not supported. These are: `sbyte, ushort, uint & ulong`.

I guess I have to support them as it is expected to use them. So I will implement them.

**Comment by friflo** [comment]
Hi,
just released v1.24.0 now supporting component fields of type: `sbyte, ushort, uint, ulong`
